File: guides/micronaut-produces-xml/java/src/main/java/example/micronaut/BookController.java
package example.micronaut;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;

@Controller("/book") // <1>
class BookController {
    @Produces(MediaType.APPLICATION_XML) // <2>
    @Get // <3>
    Book index() {
        return new Book("Building Microservices", "1491950358");
    }
}

File: guides/mn-application-aws-lambda-java11/kotlin/src/main/kotlin/example/micronaut/BookController.kt
package example.micronaut

import io.micronaut.http.annotation.Body
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Post
import java.util.UUID
import javax.validation.Valid

@Controller
open class BookController {

    @Post
    open fun save(@Valid @Body book: Book): BookSaved {
        val bookSaved = BookSaved()
        bookSaved.name = book.name
        bookSaved.isbn = UUID.randomUUID().toString()
        return bookSaved
    }
}
